microsoft/vscode
# Fix markdown cell do not re-render when re-executed

This PR fixes #151151

With this PR I added a new command (`notebook.cell.finishEdit`) that increments the cell's text model version ID when triggered, I changed the re-rendering logic from content-based to version ID-based as well. This is done to distinguish between the action of re-executing cell and stop editing cell.

Meanwhile the already-existing command (`notebook.cell.quitEdit`) now only exits the cell edit mode when triggered, it does not trigger a re-render.

Tbh I don't fully understand some parts of the changes, so let me know if I can do some things better.

## How to test
1. Create a Markdown cell with an  tag referring to a non-existing path, e.g. `<img src="myimg.png">`.
2. Execute the cell. A "broken image" icon appears.
3. Create the image at the specified path.
4. Click the cell to edit. Re-execute it (without making any changes).
